hil-test/tests/lcd_cam_i8080_async.rs

```diff
@@ -50,7 +50,11 @@ mod tests {
 
     #[test]
     async fn test_i8080_8bit(ctx: Context<'static>) {
-        let channel = ctx.dma.channel0.configure(false, DmaPriority::Priority0);
+        let channel = ctx
+            .dma
+            .channel0
+            .configure(false, DmaPriority::Priority0)
+            .into_async();
```

Comment:
Ideally I should be able to do this instead (on the chips that support it)
```suggestion
            .configure(false, DmaPriority::Priority0)
            .tx
            .into_async();
```

Reply:
You'll not need to do this if I finish implementing the rest of this. The DMA channel modes will be managed by the peripherals. Channel and likes remain public for third party drivers.

Reply:
Yes, I'm temporarily making things worse, but there's a reason for this.

Reply:
> You'll not need to do this if I finish implementing the rest of this. The DMA channel modes will be managed by the peripherals. Channel and likes remain public for third party drivers.

Ohhh I see, I think I can kinda picture what you intend here. Users won't be calling `into_async()` or `into_blocking()` on the channels, the drivers themselves will do it, during/after creation. Then all the channel interrupt APIs will be accessible via the peripheral.
This mostly makes sense I think.

(It's just a shame about the LCD_CAM thing)

Reply:
I'm not yet sure if I want `.split()` on all DMA channels (e.g. where there is a single interrupt handler) - regardless, introducing that will resolve this issue especially since the S3 has independent DMA channels.

Reply:
My thinking for this single interrupt handler business is that we should enforce that only one half of the channel can be async when a split occurs.
I currently can't envision a situation where this causes issues on C2, C3, ESP32 or S2. There's no scenario where you actually want to split the DMA channel (and have both sides be async) as most peripherals are bidirectional except SHA.

Reply:
> > Allow using the RX half and TX half in different (CPU/DMA) modes (Sometimes you only really need DMA in one half of the SPI transfer. This one might be difficult to support)
> 
> This means we can reuse the other half of the DMA channel elsewhere even on the C2/C3.

Ha, you caught me. Fair, I wasn't sure if this would be supported anyway. I suppose there's the story of 3rd party drivers, which means it should be supported regardless of what the hal itself supports.

> > I currently can't envision a situation where this causes issues on C2, C3, ESP32 or S2.
> 
> I'm not sure if this is entirely true. The I2S on ESP32 and S2 act a bit like LCD_CAM, I can see possibilities for splitting the DMA for that.

I did look into this but putting DMA aside, splitting I2S in half is difficult due to that fact that the TX and RX controls are interleaved in the same registers, so you need locks around all register accesses. (It's not a blocker I guess)

Cool, splitting needs to be supported on all chips. I'm just not too excited about having to share the one handler between the hal and the user.

Reply:
> I'm just not too excited about having to share the one handler between the hal and the user.

Yeah, me neither, but at this point I'll take an "it's not impossible".